src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/factory/generator/server42/Server42IslRttOutputFlowGenerator.java

```diff
@@ -0,0 +1,96 @@
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.switchmanager.factory.generator.server42;
+
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.actionSetDstMac;
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.actionSetOutputPort;
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.actionSetSrcMac;
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.buildInstructionApplyActions;
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.convertDpIdToMac;
+import static org.openkilda.floodlight.switchmanager.SwitchFlowUtils.prepareFlowModBuilder;
+import static org.openkilda.floodlight.switchmanager.SwitchManager.INPUT_TABLE_ID;
+import static org.openkilda.floodlight.switchmanager.SwitchManager.SERVER_42_ISL_RTT_OUTPUT_PRIORITY;
+import static org.openkilda.floodlight.switchmanager.SwitchManager.SERVER_42_ISL_RTT_REVERSE_UDP_PORT;
+import static org.openkilda.model.cookie.Cookie.SERVER_42_ISL_RTT_OUTPUT_COOKIE;
+
+import org.openkilda.floodlight.KildaCore;
+import org.openkilda.floodlight.switchmanager.factory.SwitchFlowTuple;
+import org.openkilda.floodlight.switchmanager.factory.generator.SwitchFlowGenerator;
+import org.openkilda.model.MacAddress;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import lombok.Builder;
+import net.floodlightcontroller.core.IOFSwitch;
+import org.projectfloodlight.openflow.protocol.OFFactory;
+import org.projectfloodlight.openflow.protocol.OFFlowMod;
+import org.projectfloodlight.openflow.protocol.action.OFAction;
+import org.projectfloodlight.openflow.protocol.match.Match;
+import org.projectfloodlight.openflow.protocol.match.MatchField;
+import org.projectfloodlight.openflow.types.EthType;
+import org.projectfloodlight.openflow.types.IpProtocol;
+import org.projectfloodlight.openflow.types.OFPort;
+import org.projectfloodlight.openflow.types.TransportPort;
+
+import java.util.List;
+
+public class Server42IslRttOutputFlowGenerator implements SwitchFlowGenerator {
+
+    private final KildaCore kildaCore;
+    private int server42Port;
+    private MacAddress server42MacAddress;
+
+    @Builder
+    public Server42IslRttOutputFlowGenerator(
+            KildaCore kildaCore, int server42Port, MacAddress server42MacAddress) {
+        this.kildaCore = kildaCore;
+        this.server42Port = server42Port;
+        this.server42MacAddress = server42MacAddress;
+    }
+
+    @Override
+    public SwitchFlowTuple generateFlow(IOFSwitch sw) {
+        OFFactory ofFactory = sw.getOFFactory();
+
+        List<OFAction> actions = Lists.newArrayList(
+                actionSetSrcMac(ofFactory, convertDpIdToMac(sw.getId())),
+                actionSetDstMac(ofFactory,
+                        org.projectfloodlight.openflow.types.MacAddress.of(server42MacAddress.getAddress())),
+                actionSetOutputPort(ofFactory, OFPort.of(server42Port)));
```

Comment:
also you must push server42 vlan here. Because this vlan will be used by switch 580 to send packet to server42

```
        if (server42Vlan > 0) {
            actions.add(actionPushVlan(ofFactory, EthType.VLAN_FRAME.getValue()));
            actions.add(actionReplaceVlan(ofFactory, server42Vlan));
        }
```

Reply:
As per our discussion - added.